changefeedccl: remove the default values from SHOW CHANGEFEED JOB output

sherman-grewal opened the pull request:
changefeedccl: remove the default values from SHOW
CHANGEFEED JOB output

Resolves #78420 

Currently, when a user alters a changefeed, we
include the default options in the SHOW CHANGEFEED
JOB output. In this PR we prevent the default values
from being displayed.

Release note (enterprise change): Remove the default
values from the SHOW CHANGEFEED JOB output

Jira issue: CRDB-14694

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/79325)
<!-- Reviewable:end -->

Commit:
changefeedccl: remove the default values from SHOW
CHANGEFEED JOB output

Currently, when a user alters a changefeed, we
include the default options in the SHOW CHANGEFEED
JOB output. In this PR we prevent the default values
from being displayed.

Release note (enterprise change): Remove the default
values from the SHOW CHANGEFEED JOB output
